BarrysPeanutsAkkaFlat/src/main/java/barryspeanuts/actor/PaymentActor.java

```diff
@@ -0,0 +1,142 @@
+package barryspeanuts.actor;
+
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.Customer;
+import java.util.Date;
+import java.util.UUID;
+
+public class PaymentActor extends AbstractBehavior<Object> {
+  private PaymentActor(ActorContext<Object> context) {
+    super(context);
+  }
+
+  public static Behavior<Object> create() {
+    return Behaviors.setup(PaymentActor::new);
+  }
+
+  public static Behavior<Object> behavior() {
+    return Behaviors.setup(PaymentActor::new);
+  }
+
+  @Override
+  public Receive<Object> createReceive() {
+    return newReceiveBuilder().onMessage(PaymentInfo.class, this::handlePayment).build();
+  }
+
+  private Behavior<Object> handlePayment(PaymentInfo msg) {
+    CreditCard creditCard = msg.getCreditCard();
+
+    // Get purchase state
+    getContext()
+        .getLog()
+        .info(
+            "TODO: Getting purchase state for payment according to PaymentId {}\n", msg.purchaseId);
+    double amount = msg.getPaymentAmount();
+    // Now pay
+    getContext()
+        .getLog()
+        .info(
+            "Paying with Credit Card for {} with Credit Card Number {} for the amount of {}\n",
+            creditCard.getNameOnCard(),
+            creditCard.getCreditCardNumber(),
+            amount);
+
+    // Save purchase state
+    getContext()
+        .getLog()
+        .info(
+            "TODO: Saving purchase state for payment according to PaymentId {}\n", msg.purchaseId);
+
+    // Send a payment receipt
+    CustomerActor.PaymentReceipt paymentReceipt = new CustomerActor.PaymentReceipt(msg.purchaseId);
+    ActorSystem<Object> customerActor = ActorSystem.create(CustomerActor.create(), "customerActor");
+    customerActor.tell(paymentReceipt);
+    return this;
+  }
+
+  public static class PaymentInfo {
+
+    private final UUID id;
+    private final Customer customer;
+    private final CreditCard creditCard;
+    private double paymentAmount;
+    private UUID purchaseId;
```

Comment:
How/where is this field used?

Reply:
The way the Akka Java works is to handle messages according to message type. `PaymentInfo` is a message type. I have those fields include for downstream convenience when the message is consumed.

Make sense @cretz ?

Reply:
But this field has no getter and nobody calling it